python/paddle/fluid/contrib/slim/tests/QAT_mkldnn_int8_readme.md

```diff
@@ -1,58 +1,105 @@
+
+The information about the quantization scales is being collected from three types of operators:
+
+* `fake_quantize_moving_average_abs_max` - imitates INT8 quantization of FP32 tensors, but keeps quantized output values as floats; is used before quantized operator (e.g. `conv2d`) to gather scale information for the op's input.
+* `fake_dequantize_max_abs` - imitates dequantization of INT8 tensors back into floats; it is used after quantized operator, and contains scale used for the op's weights dequantization.
+* `fake_quantize_dequantize_moving_average_abs_max` - imitates immediate quantization and dequantization; it can be used after a quantized operator to get the scale value for the op's output.
 
 Notes:
 
-* MKL-DNN and MKL are required. The performance gain can only be obtained with AVX512 series CPU servers.
-* INT8 accuracy is best on CPU servers supporting AVX512 VNNI extension.
+1. As the next steps describe, quantization will be applied later to an optimized FP32 model. It means that quantization scales for inputs and outputs of each quantized operator have to be gathered for tensors which are inputs and outputs of already optimized or fused operators. For example, if a model contains the following sequence of tensors and operators in the graph
+   ```... → input1 → conv2d → output1 → batch_norm → output2 → relu → output3 → ...```
+   and we want to quantize the `conv2d` op, then after applying FP32 optimizations the sequence will become
+   ```... → input1 → conv2d → output3 → ...```
+   and the quantization scales have to be collected for the `input1` and `outpu3` tensors in the QAT model.
+2. Quantization of the following operators is supported: `conv2d`, `depthwise_conv2d`, `mul`, `fc`, `pool2d`, `reshape2`, `transpose2`, `concat`.
+
+### Removing fake operators
+
+All the `fake_quantize_*` and `fake_dequantize_*` operators are being removed from the graph.
+
+### Dequantizing weights
+
+Weights of `conv2d` and `mul` operators are assumed to be fake-quantized (quantized, but kept as floats) in QAT models. Here, the information about the scale from `fake_dequantize_max_abs` operators is used to fake-dequantize the weights back to the full float range of values. At this moment the model becomes an unoptimized clean FP32 inference model.
+
+### Optimizing FP32 graph
+
+A series of standard optimization passes are being applied to the FP32 graph. This gives us an optimized FP32 inference model and we can proceed with INT8 quantization.
+
+### Computing weight scales
+
+After optimization fuses, the weight tensors of `conv2d` or `fc` operators are likely to have different values and require new quantization scales. The weights are static, i.e. they do not change during the inference process, and the scales can be calculated simply as a maximum of absolute values from the tensor. To improve the inference accuracy we calculate the scales for each output channel separately, getting an array of quantization scales for a weight tensor.
```

Comment:
What do you think to describe how it works?

We now introduce the notation regarding the int8 quantization process of the models. The floating-point number which is used during an INT8 inference process is a signed type of tensor. As an example, it can be considered as follows: given a set of n points in the range of the floating precision, `a^{(1)},...a^{(n)} in fp32`, find the number that is maximum or minimum of the matrix: `min_in f(x)` and `max_in f(x)`. Where `F(x)` is defined to be `f(x) = {x: -128 <= x <= 127: x in Rs}` and `f(x) = {x: 0 <= x <= 255 : x in Ru}` for output bitwidth set to 8 bits, where: Rs and Ru are for signed and unsigned values of the input. These `min_{in}` and `max_{in}` values are being set by the use of the special pre-processing step in which the re-calculating process is done.